lib/kernel/subgroups.cl

```diff
@@ -59,3 +59,53 @@ sub_group_all (int predicate)
 {
   return sub_group_reduce_min ((unsigned)predicate);
 }
+
+uint2 _CL_OVERLOADABLE
+intel_sub_group_block_read2 (const global uint *p)
+{
+  return (uint2)(p[get_sub_group_local_id ()],
+                 p[get_sub_group_local_id () + get_max_sub_group_size ()]);
+}
+
+uint8 _CL_OVERLOADABLE
+intel_sub_group_block_read8 (const global uint *p)
+{
+  return (uint8)(p[get_sub_group_local_id ()],
+                 p[get_sub_group_local_id () + get_max_sub_group_size ()],
```

Comment:
would be more readable if the `get_sub_group_local_id()` was extracted into a variable...